src/lib.rs

```diff
@@ -1688,4 +1843,50 @@ endmodule",
 
         get_inputs_and_outputs_serialized(&egraph.serialize(SerializeConfig::default()));
     }
+    #[test]
+    fn compile_rewrite_rule() {
+        let mut egraph = EGraph::default();
+        import_churchroad(&mut egraph);
+        egraph
+            .parse_and_run_program(
+                r#"
+(let v0 (Wire "v0" 4))
+; clk
+(let v1 (Wire "v1" 1))
+; out
+(let v2 (Wire "v2" 4))
+
+; cells
+; 1'1
+(let v3 (Op0 (BV 1 1)))
+; TODO not handling signedness
+(let v4 (Op1 (ZeroExtend 4) v3))
+(union v0 (Op2 (Add) v2 v4))
+; TODO: assuming 0 default for Reg
+(union v2 (Op2 (Reg 0) v1 v0))
+
+; inputs
+(let clk (Var "clk" 1))
+
+
+(IsPort "" "clk" (Input) clk)
+(union v1 clk)
+
+; outputs
+(let out v2)
+(IsPort "" "out" (Output) out)
+
+; delete wire expressions
+(delete (Wire "v0" 4))
+(delete (Wire "v1" 1))
+(delete (Wire "v2" 4))
+                "#,
+            )
+            .unwrap();
+        let serialized = egraph.serialize(SerializeConfig::default());
+        let out = AnythingExtractor.extract(&serialized, &[]);
+
+        let v1 = to_rewrite_rule_egraph_serialize(&serialized, &out, "");
+        println!("out:\n{}", v1);
```

Comment:
Can you assert that this is equal to something? It's okay if you just copy and paste in the rule that's currently printed. That'll make a brittle test, but it's better than not asserting anything at all!

Reply:
Oh - I'm not done with the rewrite-rule generation(still not done with the RHS) and the test was just there so I can run the code and see what it outputs.